# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/base/glibc/glibc23/pentium23_fix.patch
# ROCK Linux is Copyright (C) 1998 - 2006 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- ./sysdeps/unix/sysv/linux/i386/sysdep.h.orig	2003-01-08 12:38:31.000000000 +0100
+++ ./sysdeps/unix/sysv/linux/i386/sysdep.h	2003-01-08 12:41:09.000000000 +0100
@@ -35,6 +35,12 @@
 
 #ifdef __ASSEMBLER__
 
+/* __i686 is used in assembler code below as normal text within a statement.
+   When optimizing for pentium2 or pentium3, gcc defines __i686 as cpp macro
+   which destroys the code below. Glibc is only using the cpp macro __i686__,
+   so we can savely undefine __i686 here. - Clifford Wolf */
+#undef __i686
+
 /* Linux uses a negative return value to indicate syscall errors,
    unlike most Unices, which use the condition codes' carry flag.
 
